Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -780,9 +782,10 @@ void HTMLInputElement::attributeChanged(const QualifiedName& name, const AtomStr
         addToRadioButtonGroup();
         HTMLTextFormControlElement::attributeChanged(name, oldValue, newValue, attributeModificationReason);
         break;
-    case AttributeNames::checkedAttr:
+    case AttributeNames::checkedAttr: {
+        std::optional<Style::PseudoClassChangeInvalidation> checkedInvalidation;
         if (m_inputType->isCheckable())
-            invalidateStyleForSubtree();
```

Comment:
The test failure suggests that we also need to use `PseudoClassChangeInvalidation` for the default pseudo class when the checked attribute changes (there's probably a test lacking for input[type=radio] + checked attribute + :has(:default), that would show the bug)

Reply:
Yes, that is right.